src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3708,6 +3710,58 @@ private static bool TryMapParameter((SyntaxNode? Node, int Ordinal) parameterKey
             }
         }
 
+        protected virtual void ReportLambdaAttributeRudeEdits(SemanticModel oldModel,
+            SyntaxNode oldLambdaBody,
+            SemanticModel newModel,
+            SyntaxNode newLambdaBody,
+            List<RudeEditDiagnostic> diagnostics,
+            bool hadSignatureEdits,
+            CancellationToken cancellationToken)
+        {
+            var newLambda = GetLambda(newLambdaBody);
+            var oldLambda = GetLambda(oldLambdaBody);
+
+            Debug.Assert(IsNestedFunction(newLambda) == IsNestedFunction(oldLambda));
+
+            // queries are analyzed separately
+            if (!IsNestedFunction(newLambda))
+            {
+                return;
+            }
+
+            var oldLambdaSymbol = GetLambdaExpressionSymbol(oldModel, oldLambda, cancellationToken);
+            var newLambdaSymbol = GetLambdaExpressionSymbol(newModel, newLambda, cancellationToken);
+
+            var reportDiagnostic = false;
+            if (!oldLambdaSymbol.GetAttributes().SequenceEqual(newLambdaSymbol.GetAttributes()) ||
```

Comment:
>SequenceEqual [](start = 49, length = 13)

`AttributeData` does not implement equality, so I think this compares reference identity which is probably not what we want.